jetty-client/src/main/java/org/eclipse/jetty/client/HttpClient.java

```diff
@@ -179,6 +179,11 @@ public HttpClient(SslContextFactory sslContextFactory)
     public HttpClient(HttpClientTransport transport, SslContextFactory sslContextFactory)
     {
         this.transport = transport;
+        if (sslContextFactory == null)
+        {
+            sslContextFactory = new SslContextFactory(false);
+            sslContextFactory.setEndpointIdentificationAlgorithm("HTTPS");
+        }
```

Comment:
Seeing that we have required users to pass in SslContextFactory, should we do some minimal checks and warn if there is not endpoint identification algorithm set!       Perhaps provide a new constructor with a boolean to not check if people really want to avoid the warning.   We should also perhaps check for trustAll?

Reply:
@gregw there is no end to these checks... what about TLS protocol version? cipher strength? renegotiation? keystore type? etc.

I say if you provide your own `SslContextFactory` you are on your own - the default is good though.